### PR TITLE
misc: preparation for implementing MPI-5 ABI prototype

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
Split the preparation commits from https://github.com/pmodels/mpich/pull/6390

[skip warnings]

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
test:mpich/ch3/most
test:mpich/ch4/most

**Comment by hzhou** [comment]
test:mpich/ch3/most
test:mpich/ch4/most

**Comment by hzhou** [comment]
test:mpich/ch3/most
test:mpich/ch4/most

**Comment by hzhou** [comment]
test:mpich/ch3/most
test:mpich/ch4/most

File: configure.ac [hunk 1]
```diff
@@ -807,9 +807,7 @@ export FCFLAGS
 export FFLAGS
 export LDFLAGS
 export LIBS
-export MPILIBNAME
```
Comment (raffenet) [inline configure.ac:810]: Doesn't ROMIO configure use this? Or is the information passed in some other way?

Comment (hzhou) [reply configure.ac:810]: When we build ROMIO inside MPICH, we always build it as a convenience library and linked directly, right? So I don't see why it needs to know the MPILIBNAME.

Comment (raffenet) [reply configure.ac:810]: OK, I've got it now. Makes sense.

**Review by raffenet — APPROVED** [review]

**Pull request merged**